### ISSUE TITLE
New --namespace flag doesn't work for 'connect'

### ISSUE THREAD

**Issue by Poundex** [telepresenceio/telepresence#3375]
Hello,

I've just updated to Telepresence 2.16.0 in order to use the new `--namespace` flag. The goal is to use an env dump from a running pod as an EnvFile for a local service that will talk to its in-cluster peers via `telepresence connect`. The environment variables in the cluster pod all use the local (unqualified) service name, and so until now it was necessary to go through all the hostnames and append the namespace.

I thought this was going to change with the introduction of the new `--namespace` flag but it doesn't seem to be working:

```
sh-5.1$ ./telepresence-2.16.0 quit -s
sh-5.1$ ./telepresence-2.16.0 connect --namespace devs-dev
Launching Telepresence User Daemon
Launching Telepresence Root Daemon
Connected to context dev-cluster, namespace devs-dev (https://gr7.eu-west-2.eks.amazonaws.com)
sh-5.1$ http get http://devs-dev-coreplat-address-service:8080

http: LogLevel.ERROR: gaierror: [Errno -2] Name or service not known
Couldn’t resolve the given hostname. Please check the URL and try again.
```
and it still seems to be necessary to append the namespace:
```
sh-5.1$ http get http://devs-dev-coreplat-address-service.devs-dev:8080
HTTP/1.1 404 Not Found
content-length: 230
content-type: application/json
date: Thu, 05 Oct 2023 15:23:04 GMT
server: istio-envoy
x-envoy-decorator-operation: devs-dev-coreplat-address-service.devs-dev.svc.cluster.local:8080/*
x-envoy-upstream-service-time: 3

{
    "additionalData": {
        "service": "address-service"
    },
    "status": "404",
}
```

Is this supposed to be working for `connect`? If so is there configuration or something I may have missed? If not is this functionality planned for `connect`?

Thanks

**Comment by knlambert** [comment]
Hello,

What is the environment you're running Telepresence in ? It Looks like a container.

**Comment by Poundex** [comment]
It's a desktop environment with Telepresence running natively on the host Linux machine. I just switched shells for the example so I didn't have to edit all the formatting symbold out

**Comment by knlambert** [comment]
It works correctly on my Linux Mint : 

```cli
➜  ~ k get svc
NAME        TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)           AGE
redis       ClusterIP   10.43.247.77   <none>        80/TCP            132d
nginx       ClusterIP   10.43.24.216   <none>        80/TCP            50d
echo-easy   ClusterIP   10.43.29.84    <none>        80/TCP,8081/TCP   132d
➜  ~ telepresence connect --namespace blue
Connected to context default, namespace blue (xxx)
➜  ~ curl echo-easy.blue
Request served by echo-easy-85ff45d89c-lntj6

GET / HTTP/2.0

Host: echo-easy.blue
Accept: */*
User-Agent: curl/7.81.0
X-Forwarded-Proto: http
X-Request-Id: 17fd14fa-63d1-4d2f-b5ef-0d6da86fb498
➜  ~ curl echo-easy     
Request served by echo-easy-85ff45d89c-lntj6

GET / HTTP/2.0

Host: echo-easy
Accept: */*
User-Agent: curl/7.81.0
X-Forwarded-Proto: http
X-Request-Id: 92c1cd4f-4072-4125-968f-466ad1a6b801
```

What Distro do you use, and what is the result of this command when you're connected ? Mine :

```cli
➜  ~ resolvectl status                    
[...]
Link 5 (tel0)
    Current Scopes: DNS
         Protocols: -DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
Current DNS Server: 10.42.0.14
       DNS Servers: 10.42.0.14
        DNS Domain: blue.svc.cluster.local tel2-search.cluster.local ~ambassador ~app ~blue ~cluster.local ~default ~kube-node-lease ~kube-public ~kube-system
                    ~red ~tel2-search ~telepresence-compose-demo ~test ~test-ambassador
```

**Comment by Poundex** [comment]
I should've checked this first but I have now noticed that DNS queries work, even though HTTP requests don't. The DNS answer contains the fully qualified hostname (`thing.namespace.svc.cluster.local`) even when the question didn't. I'm guessing this is the issue.

Did you add those search domains by hand or did Telepresence add them for you? I'm not using systemd-resolved just resolv.conf

**Comment by knlambert** [comment]
Telepresence does it. 

If it's not using systemd-resolved, that may be that this particular use case which is broken. Can you tell me what distribution you use so we get a chance to reproduce ?

Thank you :)

**Comment by dennis-rall** [comment]
I am running it on manjaro (Arch Linux) and am facing the same problems

**Comment by Poundex** [comment]
Stock Arch for me

**Comment by dennis-rall** [comment]
Any updates on this? Still having this issue with v.2.17.0

**Comment by thallgren** [comment]
Closed by #3489
